core/tests/routes/test_routes_upload.py

```diff
@@ -0,0 +1,90 @@
+import json
+import time
+import uuid
+import random
+import pytest
+from fastapi import HTTPException
+
+
+def test_upload_memory(client):
+    # prepare test config
+    invented_key = "some-key"
+    payload = {
+        "model_name": "gpt-3.5-turbo",
+        "openai_api_key": invented_key
+    }
+
+    # act
+    response = client.put("/settings/llm/LLMOpenAIChatConfig", json=payload)
+
+    # Read fake memory
+    memory_path = "tests/mocks/sample.json"
+    with open(memory_path, "r") as f:
+        fake_memory = json.load(f)
+
+    response = client.post("/rabbithole/memory/",
+                           files={
+                               "file": ("sample.json",
+                                        fake_memory)
+                           },
+                           headers={"accept": "application/json"})
+
+    assert response.status_code == 200
```

Comment:
I think this test fails because an embedder or LLM should be set prior to uploading the memories, because of the checks about having the same embeder and embedding size